EmotiBit.cpp

```diff
@@ -3538,29 +3493,43 @@ bool EmotiBit::loadConfigFile(const String &filename) {
 	// Don't forget to change the capacity to match your JSON document.
 	// Use arduinojson.org/assistant to compute the capacity.
 	//StaticJsonBuffer<1024> jsonBuffer;
-	StaticJsonBuffer<1024> jsonBuffer;
+	StaticJsonDocument<1024> jsonDoc;
 
 	// Parse the root object
-	JsonObject &root = jsonBuffer.parseObject(file);
+	DeserializationError error = deserializeJson(jsonDoc, file);
 
-	if (!root.success()) {
+	if (error) {
 		Serial.println(F("Failed to parse config file"));
 		setupFailed("Failed to parse Config file contents");
 		return false;
 	}
 
 	size_t configSize;
 	// Copy values from the JsonObject to the Config
-	configSize = root.get<JsonVariant>("WifiCredentials").as<JsonArray>().size();
+	configSize = jsonDoc["WifiCredentials"].size(); 
 	Serial.print("Number of network credentials found in config file: ");
 	Serial.println(configSize);
 	for (size_t i = 0; i < configSize; i++) {
-		String ssid = root["WifiCredentials"][i]["ssid"] | "";
-		String userid = root["WifiCredentials"][i]["userid"] | "";
-		String username = root["WifiCredentials"][i]["username"] | "";
-		String pass = root["WifiCredentials"][i]["password"] | "";
+		String ssid = jsonDoc["WifiCredentials"][i]["ssid"].as<String>();
+		String userid = jsonDoc["WifiCredentials"][i]["userid"];
+		String username = jsonDoc["WifiCredentials"][i]["username"].as<String>();
+		String pass = jsonDoc["WifiCredentials"][i]["password"].as<String>();
 		Serial.print("Adding SSID: ");
-		Serial.print(ssid); 
+		Serial.print(ssid);
+
```

Comment:
Solution to "null" being returned instead of empty string.

Reply:
If there a reason why we have a `if` block instead of using the `|` (OR) operation as indicated in the [example](https://arduinojson.org/v6/example/config/)? (also used in the existing code)

Reply:
When config.txt is parsed, missing values are stored as a string "null" rather than an actual null value. To get around this, an if statement is used to set "null" strings to "". I didn't check what was returned before the update but the string "null" began to appear after updating to the latest version.

Reply:
I removed the "if" block. Looked a bit redundant. Instead, now we use the "OR" functionality as demonstrated in the example https://arduinojson.org/v6/api/jsonvariant/or/